src/registry/routes/helpers/get-component-fallback.js

```diff
@@ -4,28 +4,63 @@ var request = require('minimal-request');
 var url = require('url');
 var _ = require('underscore');
 
-module.exports = function(fallbackRegistryUrl, headers, component, callback) {
-  return request({
-    method: 'post',
-    url: fallbackRegistryUrl,
-    headers: _.extend({}, headers, {'host': url.parse(fallbackRegistryUrl).host}),
-    json: true,
-    body: {
-      components: [
-        component
-      ]
+module.exports = {
+  getComponent: function (fallbackRegistryUrl, headers, component, callback) {
+    return request({
+      method: 'post',
+      url: fallbackRegistryUrl,
+      headers: _.extend({}, headers, {'host': url.parse(fallbackRegistryUrl).host}),
+      json: true,
+      body: {
+        components: [
+          component
+        ]
+      }
+    }, function (err, res) {
+      if (err || !res || res.length === 0) {
+        return callback({
+          status: 404,
+          response: {
+            code: 'NOT_FOUND',
+            error: err
+          }
+        });
+      }
+
+      return callback(res[0]);
+    });
+  },
+  getComponentInfoOrPreviewFallback: function (conf, req, res, localRegistryError, component, callback) {
```

Comment:
I had some trouble coming up with good names for this and the other (`getComponent`) functions.

The `getComponent` function goes to fallback repository and makes a `POST` query for a rendered component.

The `getComponentInfoOrPreviewFallback` function repeats a `GET` query to fallback registry and was tested only with `~info` and `~preview` endpoints.